### ISSUE TITLE
Option parser doesn't handle bare --opt edge case

### ISSUE THREAD

**Issue by adigitoleo** [voldikss/vim-floaterm#176]
### Describe the bug

`b:floaterm_opts[option]` gets the incorrect value of `v:true` when running `:FloatermNew --opt` without giving an actual value. We need to make the cmdline parser throw an error for this case.

### To Reproduce

1. `:FloatermNew --name` (hit enter directly afterwards, do not type the `=SomeName` part)
2. `<C-\><C-n>`
3. `:echo b:floaterm_opts`

It's pretty harmless, but I discovered this while working on #175 . Will try to fix promptly.

**Comment by voldikss** [comment]
Oops! I even haven't noticed it before. What's your opinion on this? I propose remove that `v:true` related code in cmdline.vim. 

Also the instruction in README should amended,

```diff
- The options is formed as --key[=value],
- The options is formed as --key=value,
```

It implies that the `value` part won't be optional anymore and every argument should contains a value. For type `bool` arguments, use `--key=1` or `--key=0`.

**Comment by adigitoleo** [comment]
Yeah, I was trying to understand what the `v:true` part is for, but it will definitely be easier if we only allow arguments containing `=` character.

**Comment by adigitoleo** [comment]
So we only parse as floaterm options if it matches:
`if arg =~ '^--\S=.*$'`

**Comment by adigitoleo** [comment]
Fixed in #177
